packages/sdks/react/src/lib/useFlag.ts

```diff
@@ -0,0 +1,118 @@
+import { useEffect, useState, useRef } from 'react';
+import {
+  AllowedVariableType,
+  ChangeTypes,
+  logger,
+} from '@ninetailed/experience.js-shared';
+import { ChangesState } from '@ninetailed/experience.js';
+import { isEqual } from 'radash';
+import { useNinetailed } from './useNinetailed';
+
+export type FlagStatus = 'loading' | 'success' | 'error';
+
+export interface FlagResult<T> {
+  value: T;
+  isLoading: boolean;
```

Comment:
I wonder if maybe we want to not return `isLoading` here and have people just use the `status`.
In the `useProfile` and `useExperience` hook we expose `loading: boolean` as a property, so if we want it for consistency we should expose it with that name.

But to be honest, I'm not the biggest fan of a separate loading state if it's literally right there in the status.

Also maybe we should be clearer on the type here and say 

```
export type FlagResult<T> =
  | { status: 'loading'; value: T; error: null }
  | { status: 'success'; value: T; error: null }
  | { status: 'error'; value: T; error: Error };
```

The only thing is that then we don't have `FlagStatus`, but maybe we don't need it since we now have a discriminated union?